### ISSUE TITLE
pip installation fails

### ISSUE THREAD

**Issue by danziv94** [Deci-AI/super-gradients#2072]
### 🐛 Describe the bug

pip installation fails

to reproduce go to google.colab and in a new notebook run
```
!pip install super-gradients
```

### Versions
Python Version: 3.11.11 (main, Dec  4 2024, 08:55:07) [GCC 11.4.0]
Torch Version: 2.5.1+cu121

[super-gradients_pip_error.txt](https://github.com/user-attachments/files/18518038/super-gradients_pip_error.txt)

**Comment by R0b0To** [comment]
Colab recently upgraded the runtime to python 3.11, making the pip install fail. You need to use a different version, such as 3.10, which is already there. For example

**Set a Colab environment running 3.10**:
```
!pip install virtualenv
!virtualenv -p /usr/bin/python3.10 myenv
!source myenv/bin/activate
```
```
!myenv/bin/pip install super-gradients
```

**Comment by jVictorSA** [comment]
> Colab recently upgraded the runtime to python 3.11, making the pip install fail. You need to use a different version, such as 3.10, which is already there. For example
> 
> **Set a Colab environment running 3.10**:
> 
> ```
> !pip install virtualenv
> !virtualenv -p /usr/bin/python3.10 myenv
> !source myenv/bin/activate
> ```
> 
> ```
> !myenv/bin/pip install super-gradients
> ```

Didn't resolve for me on Colab, either.


I managed to install super-gradients locally through Conda, but using python3.8 and super-gradients v3.2.0 though.

```
conda create -n my_env python==3.8
conda activate my_env
pip install super-gradients==3.2.0
```

But, unfortunately, this version of super-gradients points to the old models repositories, and i needed to follow these instructions to get the package functioning properly:
https://github.com/Deci-AI/super-gradients/issues/2057#issuecomment-2472304864

**Comment by samerrkhann** [comment]
> Colab recently upgraded the runtime to python 3.11, making the pip install fail. You need to use a different version, such as 3.10, which is already there. For example
> 
> **Set a Colab environment running 3.10**:
> 
> ```
> !pip install virtualenv
> !virtualenv -p /usr/bin/python3.10 myenv
> !source myenv/bin/activate
> ```
> 
> ```
> !myenv/bin/pip install super-gradients
> ```

Hey,

Thanks for your lead to solve the problem. Can you perhaps share the complete code with setting up the venv. For me, it is not working after setting up the venv. I would like to see how you fixed the problem for yourself.

**Comment by R0b0To** [comment]
> Hey,
> 
> Thanks for your lead to solve the problem. Can you perhaps share the complete code with setting up the venv. For me, it is not working after setting up the venv. I would like to see how you fixed the problem for yourself.

In my case I did it locally on my pc with pytyhon 3.10.  **(I recommend you do this)**

It seems colab doesn’t natively support connecting to a local virtual environment (venv) directly. However, there are ways to work around this. 
#### Step 1 install super-gradients in the Virtual Environment
```python
!pip install virtualenv
!virtualenv -p /usr/bin/python3.10 myenv
!source myenv/bin/activate
!myenv/bin/pip install super-gradients
```
#### Step 2  Write your code in a script file (e.g., my_script.py) so it can be run using your virtual environment’s Python.
```
%%writefile my_script.py

from super_gradients.common.object_names import Models
from super_gradients.conversion import DetectionOutputFormatMode
from super_gradients.training import models

model = models.get(Models.YOLO_NAS_S, pretrained_weights="coco")
model.export("yolo_nas_s.onnx",
             output_predictions_format=DetectionOutputFormatMode.FLAT_FORMAT,
             max_predictions_per_image=20,
             confidence_threshold=0.4,
             input_image_shape=(320,320),
            )
```
>In this example I had to investigate further because the urls are not updated, If you are doing something similar the fix is here ([src/super_gradients/training/pretrained_models.py](https://github.com/Deci-AI/super-gradients/issues/2057#issuecomment-2437669724))

#### Step 3 Run the Code Using Your Virtual Environment
```
!myenv/bin/python my_script.py
```

**Comment by danziv94** [comment]
https://colab.research.google.com/drive/15KZV-Fv6BLUddxDbUUfpwswuksehGgeq?usp=drive_link 
an exmaple google colab that installs super-gradients sucesffully

code:

# Step 1: Remove Python 3.11 and its associated packages
!sudo apt-get purge -y python3.11 python3.11-minimal python3.11-dev
!sudo apt autoremove -y  # Clean up unused dependencies

# Step 2: Install Python 3.9 and necessary dependencies
!sudo apt-get update
!sudo apt-get install -y python3.9 python3.9-distutils

# Step 3: Remove any existing or broken Python symlinks
!sudo rm -f /usr/bin/python  # Remove 'python' symlink if it exists
!sudo rm -f /usr/bin/python3  # Remove 'python3' symlink if it exists

# Step 4: Create symlinks for Python 3.9 as the default Python
!sudo ln -sf /usr/bin/python3.9 /usr/bin/python3  # Point python3 to python3.9
!sudo ln -sf /usr/bin/python3.9 /usr/bin/python   # Point python to python3.9

# Step 5: Install pip for Python 3.9
!wget https://bootstrap.pypa.io/get-pip.py  # Download the get-pip.py script
!python3.9 get-pip.py  # Install pip for Python 3.9

# Step 6: Verify the installation
!python --version  # Check Python version (should be 3.9.x)
!python3 --version  # Check Python3 version (should be 3.9.x)
!pip --version  # Check pip version (should correspond to Python 3.9)

#Step 7: install super gradients
! pip install super-gradients
